finality-aleph/src/sync/substrate/verification/mod.rs

```diff
@@ -97,18 +95,26 @@ impl Display for VerificationError {
     }
 }
 
-impl<H, AP, FS> Verifier<Justification<H>> for VerifierCache<AP, FS>
+impl<AP, FS, H> Verifier<Justification<H>> for VerifierCache<AP, FS, H>
 where
-    H: SubstrateHeader<Number = BlockNumber>,
     AP: AuthorityProvider,
     FS: FinalizationInfo,
+    H: SubstrateHeader<Number = BlockNumber>,
 {
     type Error = VerificationError;
 
     fn verify(&mut self, justification: Justification<H>) -> Result<Justification<H>, Self::Error> {
         let header = &justification.header;
-        let verifier = self.get(*header.number())?;
-        verifier.verify_bytes(&justification.raw_justification, header.hash().encode())?;
-        Ok(justification)
+        match justification.clone().inner_justification {
```

Comment:
Do we need to clone here? `verify_bytes` takes `aleph_justification` via reference and we return `justification` - is compiler not smart enough to figure out we won't move it?

Reply:
It required a tiny amount of help, but now it works and one clone eliminated. Good catch, thanks.